.github/workflows/docs.yml

```diff
@@ -32,29 +50,79 @@ jobs:
           wasm32-unknown-unknown
           x86_64-unknown-linux-gnu
 
-    - name: Setup cargo-make
+    - name: setup cargo-make
       uses: davidB/rust-cargo-make@v1
 
-    - name: Install wasm-pack
+    - name: install wasm-pack
       uses: taiki-e/install-action@v2
       with:
         tool: wasm-pack
 
-    - name: Build Rust documentation
-      run: cargo doc --all --no-deps
-
-    - name: Setup bun
+    - name: setup bun
       uses: oven-sh/setup-bun@v2
       with:
         bun-version: latest
 
-    - name: Build TypeScript and Kotlin docs
+    - name: make ts docs
       run: |
         cd crypto-ffi
         cargo make docs-ts
+
+    - uses: actions/upload-artifact@v4.5.0
+      with:
+        name: tsdoc
+        path: target/doc
+        retention-days: 1
+        overwrite: true
+        include-hidden-files: true
+
+
+  ktdoc:
+    name: kotlin
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: checkout repository
+      uses: actions/checkout@v4
+
+    - uses: actions-rust-lang/setup-rust-toolchain@v1  # this implicitly caches Rust tools and build artifacts
+      with:
+        rustflags: '' # note that this is _not_ the default
+
+    - name: setup cargo-make
+      uses: davidB/rust-cargo-make@v1
+
+    - name: make kotlin docs
+      run: |
+        mkdir -p target/doc/core_crypto_ffi/bindings/kotlin
+        cd crypto-ffi
         cargo make docs-kotlin
 
-    - name: Deploy Docs
+    - uses: actions/upload-artifact@v4.5.0
+      with:
+        name: ktdoc
+        path: target/doc
+        retention-days: 1
+        overwrite: true
+        include-hidden-files: true
+
+
+  deploy:
+    name: deploy
+    runs-on: ubuntu-latest
+    if: github.repository == 'wireapp/core-crypto' && github.ref_name == 'main'
+    needs:
+      - rustdoc
+      - tsdoc
+      - ktdoc
+
+    steps:
+    - uses: actions/download-artifact@v4.1.8
+      with:
+        path: target/doc
+        merge-multiple: true
```

Comment:
Without an artifact name, wouldn't this download and merge all artifacts for the run?

Reply:
Oh, good point! Fixing.

Reply:
https://github.com/wireapp/core-crypto/pull/823/commits/73f1f2b5ff7ea02cf5c0c731df50040aef8371a6